### PR TITLE
[DOCS] Document some of the VisualScript classes

### PR DESCRIPTION
**Pull request by bojidar-bg**
Also, exposed the `ports_changed_notify` method.
Also, fixed a bug with VisualScriptCustomNode disconnecting all of its ports when the script changes.

### CODE REVIEW

**Comment by cbscribe** [comment]
One overall note: there's a mix of verb tenses in these descriptions.  For example, in some you say "Returns" and "Converts"  but then other ones say "Set" or "Remove" instead of "Sets" or "Removes".

File: doc/classes/VisualScriptCustomNode.xml [hunk 1]
```diff
@@ -109,25 +124,42 @@
 			<argument index="3" name="working_mem" type="Array">
 			</argument>
 			<description>
+				Executes the custom node's logic, returning the index of the output sequence port to use or a [String] when there is an error.
+				
+				The [code]inputs[/code] array contains the values of the input ports.
+				[code]outputs[/code] is an array whoose indices should be set to the respective outputs.
```
Comment (Noshyaar) [inline doc/classes/VisualScriptCustomNode.xml]: 'whoose' typo

File: doc/classes/VisualScriptCustomNode.xml [hunk 2]
```diff
@@ -109,25 +124,42 @@
...
+				When returning, you can mask the returned value with one of the STEP_* constants.
 			</description>
 		</method>
 	</methods>
 	<constants>
 		<constant name="START_MODE_BEGIN_SEQUENCE" value="0">
+			The start mode used the first time when [method _step] is called.
 		</constant>
 		<constant name="START_MODE_CONTINUE_SEQUENCE" value="1">
+			The start mode used when [method _step] is called after comming back from a STEP_PUSH_STACK_BIT.
```
Comment (Noshyaar) [inline doc/classes/VisualScriptCustomNode.xml]: comming

File: doc/classes/VisualScriptDeconstruct.xml [hunk 3]
```diff
@@ -1,8 +1,10 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <class name="VisualScriptDeconstruct" inherits="VisualScriptNode" category="Core" version="3.0.alpha.custom_build">
 	<brief_description>
+		A Visual Script node which deconstructs a base type instance into its parts.
```
Comment (StraToN) [inline doc/classes/VisualScriptDeconstruct.xml]: Grammatically correct, but you use both "which" and "that" in similar sentences from different doc files (eg. VisualScriptEmitSignal). I'd advise to choose one.

File: doc/classes/VisualScriptEmitSignal.xml [hunk 4]
```diff
@@ -1,8 +1,10 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <class name="VisualScriptEmitSignal" inherits="VisualScriptNode" category="Core" version="3.0.alpha.custom_build">
 	<brief_description>
+		A Visual Script node that emits a specified signal.
```
Comment (StraToN) [inline doc/classes/VisualScriptEmitSignal.xml]: For reference to previous remark : that or which.

**Review by StraToN — CHANGES_REQUESTED** [review]

**Comment by bojidar-bg** [comment]
Did the requested changes, should I also follow https://github.com/godotengine/godot-docs/issues/501?

File: doc/classes/VisualScriptDeconstruct.xml (hunk 3 above)
Comment (bojidar-bg) [reply doc/classes/VisualScriptDeconstruct.xml]: Let's say I will stay with "which".

**Review by StraToN — APPROVED** [review]

**Comment by NathanLovato** [comment]
I think it's good for now. Alpha 2 is around the corner, and we'll prob still need some more discussion and experimentation to standardize the VS classref. Having your docs will make it a lot easier to improve them later if need be.

**Pull request merged**